shared/src/api/client/mainthreadAPI.ts

```diff
@@ -28,23 +30,45 @@ export const initMainThreadAPI = (
     )
 
     // Workspace
-    subscription.add(
+    sub.add(
         roots.subscribe(rs => {
             // eslint-disable-next-line @typescript-eslint/no-floating-promises
             ext.syncRoots(rs || [])
         })
     )
-
-    subscription.add(
+    sub.add(
         versionContext.subscribe(ctx => {
             // eslint-disable-next-line @typescript-eslint/no-floating-promises
             ext.syncVersionContext(ctx)
         })
     )
 
+    // Commands
+    let nextCmdHandle = 1
+    const cmds = new Map<number, Unsubscribable>()
```

Comment:
We generally avoid abbreviations when possible in our TS codebase (same comment for `sub` -> `subscription`, `cmdRegistry` -> `commandRegistry`). I realize this may be undocumented convention right now - I'll add it to the TS section in the handbook.

```suggestion
    const commands = new Map<number, Unsubscribable>()
```